connectors/byoc.py

```diff
@@ -154,63 +153,135 @@ def _create_object(self, doc_source):
             doc_source["_source"],
             bulk_options=self.bulk_options,
             language_code=self.language_code,
-            analysis_icu=self.analysis_icu,
         )
 
 
-class SyncJob:
-    def __init__(self, connector_id, elastic_client):
-        self.connector_id = connector_id
-        self.client = elastic_client
-        self.created_at = datetime.now(timezone.utc)
-        self.completed_at = None
-        self.job_id = None
-        self.status = None
+class SyncJobIndex(ESIndex):
+    READ_ONLY_FIELDS = ("connector", "trigger_method", "created_at")
 
-    @property
-    def duration(self):
-        if self.completed_at is None:
-            return -1
-        msec = (self.completed_at - self.created_at).microseconds
-        return round(msec / 9, 2)
+    def __init__(self, elastic_config):
+        super().__init__(index_name=JOBS_INDEX, elastic_config=elastic_config)
 
-    async def start(self, trigger_method=JobTriggerMethod.SCHEDULED):
-        self.status = JobStatus.IN_PROGRESS
+    def build_docs_query(self, connector_ids):
```

Comment:
we should make the function name more specific and descriptive.

Reply:
I agree with it, however I'm following @vidok's recommendation here to inherit from ESIndex and implement the methods - I would even probably move it to a different class - something like QueryBuilder that is usable like:

```
query = JobsIndex.QueryBuilder()
    .with_native_service_types(native_service_types)
    .build()

## OR

query = JobsIndex.QueryBuilder()
    .with_native_service_types(native_service_types)
    .with_connector_ids(connector_ids)
    .build()
```

For now I'd say it's out of scope of PR and will be addressed later.